test/1_stdlib/Character.swift

```diff
@@ -304,5 +304,20 @@ UnicodeScalarTests.test("Comparable") {
   expectTrue("B" >= CharA)
 }
 
+UnicodeScalarTests.test("forall x: UnicodeScalar . x == UnicodeScalar(String(x))") {
+  expectEqualFunctionsForDomain(baseScalars,
+                                { $0 },
+                                { UnicodeScalar(String($0))! })
```

Comment:
Please follow the library coding style (break the line after the open paren, and indent the continuation line two more spaces).